### PR TITLE
Fix domain and group disassociation when toggling from Black/Whitelis…

### PR DESCRIPTION
**Pull request by trillaxe**
…t page

Signed-off-by: trillaxe <trillax@tutanota.com>

**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`

`{Please ensure that your pull request is for the 'devel' branch!}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [ ] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [X] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**

Fix the domains group associations unselected when toggling status (Enable|Disable) from
the Blacklist and Whitelist page.

**How does this PR accomplish the above?:**

By only deleting then inserting the group associations (to update it) if the groups array parameter is set.

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

**Comment by trillaxe** [comment]
Hopefully i\m doing this right, pretty new to making pull requests.

**Comment by dschaper** [comment]
You're doing just fine!

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/domains-group-assigment-is-not-preserved-between-dis-re-enabling-via-web-ui-black-whitelist/28812/9

**Comment by dschaper** [comment]
I think the whitespace was causing some issues, so I took that all out. Doesn't fix the github display but I concur that the PR just adds 2 lines and does not remove blocks.

**Comment by trillaxe** [comment]
I was looking at the code in **groups-domains.js** and thought i can make a simple improvement,
since the group assignment is only shown in Domains page not in Blacklist nor Whitelist according to the comment on the first image
groups-domains-multiselect
Maybe this can be changed from
editDomain_before
To this? it would make my first commit unnecessary, though it is harmless.
editDomain_after
In Blacklist or Whitelist page the "groups" variable is undefined since group assignment is not included and it cannot get the value of multiselect (undefined), therefore it is not set in the request.
I don't know if some browsers set undefined variables in post request's data, if so, then this improvement would be a better choice, what do you guys think?

Another thing is, **rowData** variable can be used to update row data upon successful domain edit, i can make another pull request on that.

**Comment by dschaper** [comment]
As long as the improvements are on topic for the PR and don't wander to other things, then it's all great to have! I like to have smaller, single issue PRs that can be quickly digested and approved instead of larger ones that get to be a bit heavy. 

Soon GitHub will have dependent PRs so you can set up a few small, related PRs and only allow them to be merged when the previous ones are already merged.

**Comment by dschaper** [comment]
My whitespace commit should have been it's own PR but I was hoping it would fix the GitHub diff confusion on this PR.

**Comment by trillaxe** [comment]
The "groups" variable change is very related, it is the reason why this problem exists in the first place, most likely someone forgot to change it. When i tried pushing a new commit, it said that i first need to do a git pull, so i did, not sure exactly that's what i was suppose to do?

**Comment by dschaper** [comment]
Yep, that's what you needed to do. You made a commit to a branch and pushed it, then I changed that branch with an additional commit (the whitespace). Your local branch was then out of date and you needed to pull my commit down and apply it locally. Then everyone had the same changes. `git` noticed that you didn't have all the commits in the history and warned that you needed to be in sync before it was safe to say your later changes could be used.

TLDR, everything is as it should be.

**Comment by trillaxe** [comment]
Actually, yea you're right, the "groups" variable change should probably be on it's own PR since it turns out it is also in **groups-adlists.js** aswell as **groups-clients.js**. Well it's a learning experience for me i'll pick it quickly, so what do i do to remove the last 2 commits? stackoverflow recommends that i do a git revert?

**Comment by XhmikosR** [comment]
https://github.com/pi-hole/AdminLTE/pull/1172/files?w=1

**Comment by dschaper** [comment]
Ready for review?

**Comment by trillaxe** [comment]
Sorry am i suppose to request a review?

**Comment by dschaper** [comment]
>Another thing is, rowData variable can be used to update row data upon successful domain edit, i can make another pull request on that.

I didn't know if there was more work to be done on this PR or if things were set.

**Comment by trillaxe** [comment]
well i didn't know :smile:
thank you for putting up with me, i've learned quite a bit from this pull request, so thank you!

**Comment by dschaper** [comment]
There wasn't any "putting up" needed. Thanks for the contribution and we hope you will continue to contribute in the future! :)
